runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1549,11 +1549,21 @@ TR_VectorAPIExpansion::unboxNode(TR::Node *parentNode, TR::Node *operand, vapiOb
    bool parentScalarized;
    bool parentVectorizedOrScalarized = isVectorizedOrScalarizedNode(parentNode, elementType, bitsLength,
                                                                     parentType, parentScalarized);
+   int32_t elementSize = OMR::DataType::getSize(elementType);
+   int32_t numLanes = bitsLength/8/elementSize;
+   TR::VectorLength vectorLength = OMR::DataType::bitsToVectorLength(bitsLength);
+   TR::ILOpCodes maskLoadOpCode;
+   TR::ILOpCodes maskConv;
 
    if ((operandObjectType != Vector && operandObjectType != Mask) ||
-        elementType != TR::Int8 ||
-        bitsLength != 128 ||
-        parentScalarized)  // TODO: support unboxing into scalars
+       elementType != TR::Int8 ||   // TODO: support more types (needs support in createClassesForBoxing)
+       bitsLength != 128 ||
```

Comment:
any reason to limit this to 128-bit?

Reply:
Yes, we need to create class objects for all other types. I would like to give it a thought on what's the best way to do it.

Reply:
The TODO comment explains why only 128-bit.